### PR TITLE
Gensam

### PR DESCRIPTION
**Pull request by sima-r**

### CODE REVIEW

File: gensamupload/gensamupload.py [hunk 1]
```diff
@@ -210,40 +222,88 @@ def main(runid, demultiplexdir, logdir, inputdir, samplesheetname, regioncode, l
...
+
                 sftp.put(fastqR1_src, fastqR1_trgt)
                 sftp.put(fastqR2_src, fastqR2_trgt)
-                                
+                
+                if sftp.exists( fastqR1_trgt):
+                    log.write(writelog("Uploading_info", f'{fastqR1_trgt} exists in sftp.'))
+                    report_data.append('uploaded')
+                else:
+                    log.write(writelog("Uploading_info", f'{fastqR1_trgt} dose exist in sftp.'))
```
Comment (pbiology) [inline gensamupload/gensamupload.py:246]: typo: does

File: gensamupload/gensamupload.py [hunk 2]
```diff
@@ -210,40 +222,88 @@ def main(runid, demultiplexdir, logdir, inputdir, samplesheetname, regioncode, l
...
+                    report_data.append('not uploaded')
+                if sftp.exists(fastqR2_trgt):
+                    report_data.append('uploaded')
+                    log.write(writelog("Uploading_info", f'{fastqR2_trgt} exists in sftp.'))
+                else:
+                    log.write(writelog("Uploading_info", f'{fastqR2_trgt} dose exist in sftp.'))
```
Comment (pbiology) [inline gensamupload/gensamupload.py:252]: Typo: does

File: gensamupload/gensamupload.py [hunk 3]
```diff
@@ -210,40 +222,88 @@ def main(runid, demultiplexdir, logdir, inputdir, samplesheetname, regioncode, l
...
+                    report_data.append('not uploaded')
+            else:
+                report_data.append('--no-upload')
```
Comment (pbiology) [inline gensamupload/gensamupload.py:255]: Why twice?

File: gensamupload/gensamupload.py [hunk 4]
```diff
@@ -210,40 +222,88 @@ def main(runid, demultiplexdir, logdir, inputdir, samplesheetname, regioncode, l
...
+                report_data.append('--no-upload')
+            
+        else:
+            report_data.append('R1/R2 not found')
+            report_data.append('R1/R2 not found')
```
Comment (pbiology) [inline gensamupload/gensamupload.py:260]: Also duplicated row

File: gensamupload/gensamupload.py [hunk 5]
```diff
@@ -210,40 +222,88 @@ def main(runid, demultiplexdir, logdir, inputdir, samplesheetname, regioncode, l
...
+            
             if not no_upload:
                 sftp.put(fasta_src, fasta_trgt)
-
+                if sftp.exists(fasta_trgt):
+                    report_data.append('uploaded')
+                    log.write(writelog("Uploading_info", f'{fasta_trgt} exists in sftp.'))
+                else:
+                    report_data.append('not uploaded')
+                    log.write(writelog("Uploading_info", f'{fasta_trgt} dose not exist in sftp.'))
```
Comment (pbiology) [inline gensamupload/gensamupload.py:275]: typo: does

File: nextseq_cronscript.py [hunk 6]
```diff
@@ -125,25 +131,32 @@ def upload_fastq(hcp_paths,hcpm,logger):
...
-        pass
-    process.stdout.close()
-
+def gensam_upload(sshkey,run):
+    #logsub = open("~/inbox/gensam.log", "w") 
+    #cmd = ["gensamupload/gensamupload.py", "-r", run, "--sshkey-password", args.sshkey]
+    cmd = ["python", "/apps/bio/repos/sars-cov-2-typing/gensamupload/gensamupload.py", "-r", run, "--sshkey-password", sshkey, "--no-mail", "--no-upload"]
+    print(cmd)
+    process1 = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    print(process1.stdout)
```
Comment (pbiology) [inline nextseq_cronscript.py:140]: All these print statements should perhaps be removed?

File: tools/emailer.py [hunk 7]
```diff
@@ -31,7 +31,7 @@ def email_micro(subject, body):
     msg['Subject'] = f'{subject}'
     msg['From'] = "clinicalgenomics@gu.se"
     msg['To'] = ["johan.ringlander@vgregion.se", "josefin.olausson@vgregion.se"]
-    msg['Cc'] = "anders.lind.cgg@gu.se"
+    msg['Cc'] = ["anders.lind.cgg@gu.se", "sima.rahimi@gu.se"]
```
Comment (pbiology) [inline tools/emailer.py:34]: This whole thing could be moved to config file. Easier to add new e-mails

File: gensamupload/gensamupload.py [hunk 8]
```diff
@@ -199,23 +199,20 @@ def main(runid, demultiplexdir, logdir, inputdir, samplesheetname, regioncode, l
             #add sample names to dict
                 uploads_dict[row[0]] = ''
     
-    #craete a report file
-    report = open( gensamcsvdir + '/reports/' + runid + '_gensam_upload.csv', 'w')
-    header = ['SampleID','fastq1', 'fastq2', 'fasta', 'upload_date']
+    #Open the report file 
+    report = open( gensamcsvdir + '/reports/' + 'Samples_to_gensam.csv', 'a')
+    #header = ['SampleID','criterion','RunID','sequencing_week','fastq1','fastq2','fasta','upload_date','upload_week']
```
[the remark this answers is not in the document]
Comment (sima-r) [reply gensamupload/gensamupload.py:204]: This is not used anywhere, just left it to see how the header looks like in the file. Could be deleted!

File: gensamupload/gensamupload.py [hunk 9]
```diff
@@ -376,19 +376,20 @@ def sample_sheet(sspath):
...
             continue
-
-        #Skip controls
-        if sample_name.startswith(( 'PosCtrl', 'PosKon', 'NegCtrl', 'NegKon')):
-            continue
         else:
             #populate a dictionary with samples and their selection criteria value
             sample_id = sample_name.split('_')[0]
             data[sample_id] = sample_criteria
+
```
Comment (pbiology) [inline gensamupload/gensamupload.py:392]: Isn't the if/else statements messed up here?
The else statement now is paired with the `if runtype == '00'`, right?

**Review by pbiology — CHANGES_REQUESTED** [review]

File: gensamupload/gensamupload.py (hunk 9 above)
Comment (sima-r) [reply gensamupload/gensamupload.py:392]: So those samples that are not controls or whatever we put into that list and are not  rerun samples will satisfy the else statement.

**Review by pbiology — APPROVED** [review]

**Pull request merged**